### PR TITLE
fix tracebacks in terminal from failures in `.robot` tests

### LINKED ISSUE #266: Irrelevant stack trace with plugin internals
When a robot test is run using this plugin and fails, a long stack trace is printed.

### Example test
```robot
*** Test Cases ***
Foo
    Fail
```

### Command line
```
python -m pytest tests/foo.robot
```

### Stack trace
```
================================== FAILURES ===================================
_____________________________________ Foo _____________________________________
.venv\Lib\site-packages\_pytest\runner.py:341: in from_call
   result: Optional[TResult] = func()
.venv\Lib\site-packages\_pytest\runner.py:241: in <lambda>
   lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
.venv\Lib\site-packages\pluggy\_hooks.py:513: in __call__
   return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
.venv\Lib\site-packages\pluggy\_manager.py:120: in _hookexec
   return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
.venv\Lib\site-packages\pluggy\_callers.py:139: in _multicall
   raise exception.with_traceback(exception.__traceback__)
.venv\Lib\site-packages\pluggy\_callers.py:122: in _multicall
   teardown.throw(exception)  # type: ignore[union-attr]
.venv\Lib\site-packages\_pytest\threadexception.py:87: in pytest_runtest_call
   yield from thread_exception_runtest_hook()
.venv\Lib\site-packages\_pytest\threadexception.py:63: in thread_exception_runtest_hook
   yield
.venv\Lib\site-packages\pluggy\_callers.py:122: in _multicall
   teardown.throw(exception)  # type: ignore[union-attr]
.venv\Lib\site-packages\_pytest\unraisableexception.py:90: in pytest_runtest_call
   yield from unraisable_exception_runtest_hook()
.venv\Lib\site-packages\_pytest\unraisableexception.py:65: in unraisable_exception_runtest_hook
   yield
.venv\Lib\site-packages\pluggy\_callers.py:122: in _multicall
   teardown.throw(exception)  # type: ignore[union-attr]
.venv\Lib\site-packages\_pytest\logging.py:850: in pytest_runtest_call
   yield from self._runtest_for(item, "call")
.venv\Lib\site-packages\_pytest\logging.py:833: in _runtest_for
   yield
.venv\Lib\site-packages\pluggy\_callers.py:122: in _multicall
   teardown.throw(exception)  # type: ignore[union-attr]
.venv\Lib\site-packages\_pytest\capture.py:878: in pytest_runtest_call
   return (yield)
.venv\Lib\site-packages\pluggy\_callers.py:122: in _multicall
   teardown.throw(exception)  # type: ignore[union-attr]
.venv\Lib\site-packages\pytest_robotframework\_internal\pytest\plugin.py:582: in pytest_runtest_call
   yield
.venv\Lib\site-packages\pluggy\_callers.py:122: in _multicall
   teardown.throw(exception)  # type: ignore[union-attr]
.venv\Lib\site-packages\_pytest\skipping.py:257: in pytest_runtest_call
   return (yield)
.venv\Lib\site-packages\pluggy\_callers.py:103: in _multicall
   res = hook_impl.function(*args)
.venv\Lib\site-packages\_pytest\runner.py:183: in pytest_runtest_call
   raise e
.venv\Lib\site-packages\_pytest\runner.py:173: in pytest_runtest_call
   item.runtest()
.venv\Lib\site-packages\pytest_robotframework\_internal\pytest\robot_file_support.py:126: in runtest
   ).run(
.venv\Lib\site-packages\robot\running\bodyrunner.py:60: in run
   raise ExecutionFailures(errors)
E   robot.errors.ExecutionFailures: AssertionError
=========================== short test summary info ===========================
FAILED tests/foo.robot::Foo - robot.errors.ExecutionFailures: AssertionError
============================== 1 failed in 0.70s ==============================
```

### Expected behaviour
A short traceback that includes the original exception (in this case, the `fail` in `BuiltIn.py`), but not any pytest / plugin internals.

### PR DESCRIPTION
**Pull request by KotlinIsland**
fixes #266

### CODE REVIEW

File: pytest_robotframework/_internal/pytest/robot_file_support.py [hunk 1]
```diff
@@ -25,13 +28,32 @@
...
+    def _get_failure(
+        self, exc_type: type[BaseException], exc_value: BaseException, exc_tb: Never, context: Never
+    ):
+        result = super()._get_failure(exc_type, exc_value, exc_tb, context)  # pyright: ignore[reportUnknownMemberType]
+        if result:
+            result.error = exc_value  # pyright: ignore[reportAttributeAccessIssue]
+        return result
+
+
+librarykeywordrunner.StatusReporter = StatusReporterButWorksProperly
```
Comment (KotlinIsland) [inline pytest_robotframework/_internal/pytest/robot_file_support.py:54]: Is this just `patch_method` with extra steps?

Comment (KotlinIsland) [reply pytest_robotframework/_internal/pytest/robot_file_support.py:54]: Even if it is, I think it's worth it, status reporter discarding the exception has always been really frustrating.

Comment (DetachHead) [reply pytest_robotframework/_internal/pytest/robot_file_support.py:54]: we don't want to patch every status reporter tho right, just a specific one?

File: pytest_robotframework/_internal/pytest/robot_file_support.py [hunk 2]
```diff
@@ -137,3 +158,15 @@ def teardown(self):
...
+    def repr_failure(
+        self, excinfo: ExceptionInfo[BaseException], style: _TracebackStyle | None = None
+    ) -> str | TerminalRepr:
+        if isinstance(excinfo.value, ExecutionFailures):
+            error = cast(BaseException, excinfo.value._errors[0].error)  # pyright: ignore[reportPrivateUsage, reportUnknownMemberType]
+            if error.__traceback__:
+                return super().repr_failure(
+                    ExceptionInfo((type(error), error, error.__traceback__), _ispytest=True), style
+                )
+        return super().repr_failure(excinfo, style)
```
Comment (DetachHead) [inline pytest_robotframework/_internal/pytest/robot_file_support.py:172]: erm

File: pytest_robotframework/_internal/pytest/robot_file_support.py (hunk 1 above)
Comment (DetachHead) [reply pytest_robotframework/_internal/pytest/robot_file_support.py:53]: i guess just use `patch_method` then

Comment (KotlinIsland) [reply pytest_robotframework/_internal/pytest/robot_file_support.py]: dome?

File: pytest_robotframework/_internal/pytest/robot_file_support.py [hunk 3]
```diff
@@ -20,18 +23,37 @@
...
+@patch_method(StatusReporter)
+def _get_failure(  # pyright: ignore[reportUnusedFunction]  # noqa: PLR0917
+    og: Callable[[StatusReporter, type[BaseException], BaseException, Never, Never], object],
+    self: StatusReporter,
+    exc_type: type[BaseException],
+    exc_value: BaseException,
+    exc_tb: Never,
+    context: Never,
+):
+    result = og(self, exc_type, exc_value, exc_tb, context)
```
Comment (KotlinIsland) [inline pytest_robotframework/_internal/pytest/robot_file_support.py:51]: what is this doing?

File: pytest_robotframework/_internal/pytest/robot_file_support.py [hunk 4]
```diff
@@ -137,3 +159,28 @@ def teardown(self):
     @override
     def reportinfo(self) -> tuple[PathLike[str] | str, int | None, str]:
         return (self.path, None if self.line_number is None else self.line_number - 1, self.name)
+
+    @override
+    def repr_failure(
+        self, excinfo: ExceptionInfo[BaseException], style: _TracebackStyle | None = None
+    ) -> str | TerminalRepr:
+        if isinstance(excinfo.value, ExecutionFailures):
+            error = cast(BaseException, excinfo.value._errors[0].error)  # pyright: ignore[reportPrivateUsage, reportUnknownMemberType]
```
Comment (KotlinIsland) [inline pytest_robotframework/_internal/pytest/robot_file_support.py:168]: last one?

File: pytest_robotframework/_internal/pytest/robot_file_support.py [hunk 5]
```diff
@@ -137,3 +159,28 @@ def teardown(self):
...
+            if isinstance(error, RobotError) or not error.__traceback__:
+                return Huh(excinfo.value)
+            return super().repr_failure(ExceptionInfo[BaseException].from_exception(error), style)
+        return super().repr_failure(excinfo, style)
+
+
+@dataclasses.dataclass(eq=False)
+class Huh(TerminalRepr):
```
Comment (KotlinIsland) [inline pytest_robotframework/_internal/pytest/robot_file_support.py:176]: `RobotToiletRepr`

**Pull request merged**